File: mnemosyne/mnemosyne/embedded_in_C/review_wdgt.py
#
# review_wdgt.py <Peter.Bienstman@UGent.be>
#

import _review_wdgt as _
from mnemosyne.libmnemosyne.ui_components.review_widget import ReviewWidget


class ReviewWdgt(ReviewWidget):

    def set_question_box_visible(self, is_visible):
        _.set_question_box_visible(is_visible)
        
    def set_answer_box_visible(self, is_visible):
        _.set_answer_box_visible(is_visible)
        
    def set_question_label(self, text):
        _.set_question_label(text.encode("utf-8"))
        
    def set_question(self, text):
        _.set_question(text.encode("utf-8"))
        
    def set_answer(self, text):
        _.set_answer(text.encode("utf-8"))
        
    def clear_question(self): 
        _.clear_question()
        
    def clear_answer(self): 
        _.clear_answer ()
        
    def update_show_button(self, text, is_default, is_enabled): 
        _.update_show_button(text.encode("utf-8"), is_default, is_enabled)

    def set_grades_enabled(self, is_enabled):
        _.set_grades_enabled(is_enabled)
    
    def set_grade_enabled(self, grade, is_enabled):
        _.set_grade_enabled(is_enabled)
    
    def set_default_grade(self, grade):
        _.set_default_grade(grade)
        
    def set_grades_title(self, text): 
        _.set_grades_title(text.encode("utf-8"))
            
    def set_grade_text(self, grade, text): 
        _.set_grade_text(grade, text.encode("utf-8"))
            
    def set_grade_tooltip(self, grade, text): 
        _.set_grade_tooltip(grade, text.encode("utf-8"))

    def update_status_bar_counters(self):
        _.update_status_bar_counters()

    def repaint_now(self):
        _.repaint_now()

File: mnemosyne/mnemosyne/embedded_in_C/review_wdgt.c
//
// review_wdgt.c <Peter.Bienstman@UGent.be>
//
//   Template, to be replaced by functions which actually do something useful
//   in a specific client.
//

#include <stdio.h>

void review_wdgt_set_question_box_visible(int is_visible)
{
  printf("set_question_box_visible: %d\n", is_visible);
}

        
void review_wdgt_set_answer_box_visible(int is_visible)
{
  printf("set_answer_box_visible: %d\n", is_visible);
}        


void review_wdgt_set_question_label(char* text)
{
  printf("set_question_label: %s\n", text);
}
        

void review_wdgt_set_question(char* text)
{
  printf("set_question: %s\n", text);
}        


void review_wdgt_set_answer(char* text)
{
  printf("set_answer: %s\n", text);
}
        

void review_wdgt_clear_question() 
{
  printf("clear_question\n");
}        


void review_wdgt_clear_answer() 
{
  printf("clear_answer\n");
}        


void review_wdgt_update_show_button(char* text, int is_default, int is_enabled) 
{
  printf("update_show_button: %s %d %d\n", text, is_default, is_enabled);
}


void review_wdgt_set_grades_enabled(int is_enabled)
{
  printf("set_grades_enabled: %d\n", is_enabled);
}    


void review_wdgt_set_grade_enabled(int grade, int is_enabled)
{
  printf("set_grade_enabled: %d %d\n", grade, is_enabled);
}    


void review_wdgt_set_default_grade(int grade)
{
  printf("set_default_grade: %d\n", grade);
}        


void review_wdgt_set_grades_title(char* text) 
{
  printf("set_grades_title: %s\n", text);
}            


void review_wdgt_set_grade_text(int grade, char* text) 
{
  printf("set_grade_text: %d %s\n", grade, text);
}            


void review_wdgt_set_grade_tooltip(int grade, char* text) 
{
  printf("set_grade_tooltip: %d %s\n", grade, text);
}


void review_wdgt_update_status_bar(char* message)
// Should also update the counters.
{
  printf("update_status_bar: %s\n", message);
}
